video2dataset/subsamplers/optical_flow_subsampler.py

```diff
@@ -49,18 +91,23 @@ def __call__(self, frame1, frame2):
         Returns:
             numpy.ndarray: The computed optical flow.
         """
-        frame1, frame2 = self.preprocess(frame1), self.preprocess(frame2)
-        return cv2.calcOpticalFlowFarneback(
-            frame1,
-            frame2,
-            None,
-            self.pyr_scale,
-            self.levels,
-            self.winsize,
-            self.iterations,
-            self.poly_n,
-            self.poly_sigma,
-            self.flags,
+        frame1, scaling_factor = self.preprocess(frame1)
+        frame2, _ = self.preprocess(frame2)
+
+        return (
+            cv2.calcOpticalFlowFarneback(
+                frame1,
+                frame2,
+                None,
+                self.pyr_scale,
+                self.levels,
+                self.winsize,
+                self.iterations,
+                self.poly_n,
+                self.poly_sigma,
+                self.flags,
+            )
+            * scaling_factor
```

Comment:
this is the confirmed way to go about optical flow on downscaled images right? Just making sure

Reply:
maybe andreas can review to confirm? But this is what we discussed in our thread, maintain the aspect ratio but ensure one side is a certain length and rescale accordingly